shared/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -43,6 +47,11 @@
      */
     private static final String TRANSACTION_DELIMITER = "#transaction.";
 
+    /**
+     * This is appended to the end of the Primary Segment Name, followed by secondary id.
+     */
+    private static final String SECONDARY_ID_DELIMITER = "#secondary.";
```

Comment:
Why is "secondary" needed? Isn't "#" enough?

Reply:
#secondary makes it a unique delimiter and follows the same naming convention that transaction delimiter does.

Reply:
Ok, can we make this "#epoch" then?

Reply:
sure